From d851fdd768b27c02b3fb786fd0987faddd279ece Mon Sep 17 00:00:00 2001
From: Ben Beasley <code@musicinmybrain.net>
Date: Tue, 9 Jun 2026 08:48:33 -0700
Subject: [PATCH] PR #2071: Include immintrin.h instead of bmi2intrin.h

Imported from GitHub PR https://github.com/abseil/abseil-cpp/pull/2071

GCC 16 produces an error when including `bmi2intrin.h` directly.

Merge 8ab6707df8ca3ea7263ae79930668de429ea2460 into acc35c318c4248b05ad85f6b7bbffd3c09bdfea7

Merging this change closes #2071

COPYBARA_INTEGRATE_REVIEW=https://github.com/abseil/abseil-cpp/pull/2071 from musicinmybrain:bmi2intrin 8ab6707df8ca3ea7263ae79930668de429ea2460
PiperOrigin-RevId: 929220314
Change-Id: I0ac97e2e5c1016363579026ec8e7b4786e6531bd
---
 absl/container/internal/raw_hash_set.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/absl/container/internal/raw_hash_set.h b/absl/container/internal/raw_hash_set.h
index b99d987f..7119e103 100644
--- a/absl/container/internal/raw_hash_set.h
+++ b/absl/container/internal/raw_hash_set.h
@@ -230,9 +230,9 @@
 #include <ranges>  // NOLINT(build/c++20)
 #endif
 
-#ifdef __BMI2__
-#include <bmi2intrin.h>
-#endif  // __BMI2__
+#if defined(__i386__) || defined(__x86_64__)
+#include <immintrin.h>
+#endif
 
 namespace absl {
 ABSL_NAMESPACE_BEGIN
-- 
2.56.0

